### PR TITLE
IEP-540 Not opening Aplication Size Analysis on esp-idf 4.4 and higher

### PR DESCRIPTION
**Pull request by sigmaaa**
Application Size Analysis is now working with all versions of the esp-idf.
Tested on Esp-idf 4.2, 4.3, 4.4 and 5.0. Windows10 and macOS. 

also works with different targets:
esp32 -
Screenshot 2021-12-10 at 9 50 16

esp32c3 with single diram column:
Screenshot 2021-12-10 at 9 52 31

for ESP-IDF 4.4 and higher, in the details, columns Other and Total have been changed to Ram Total and Flash Total accordingly

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java [hunk 1]
```diff
@@ -4,37 +4,125 @@
...
 	//esp32-s2 specific
-	String USED_DIRAM = "used_diram"; //$NON-NLS-1$
-	String AVAILABLE_DIRAM = "available_diram"; //$NON-NLS-1$
-	String USED_DIRAM_RATIO = "used_diram_ratio"; //$NON-NLS-1$
+	public static String USED_DIRAM = "used_diram"; //$NON-NLS-1$
+	public static String AVAILABLE_DIRAM = "available_diram"; //$NON-NLS-1$
+	public static String USED_DIRAM_RATIO = "used_diram_ratio"; //$NON-NLS-1$
 	
+	static {
+		String versing = getEspIdfVersion();
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java:56]: typo `versing` >> version

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java [hunk 2]
```diff
@@ -4,37 +4,125 @@
...
+	{
+		String exportCmdOp = ""; //$NON-NLS-1$
+		ProcessBuilderFactory processRunner = new ProcessBuilderFactory();
+		try
+		{
+			IStatus status = processRunner.runInBackground(arguments, Path.ROOT, env);
+			if (status == null)
+			{
+				Logger.log(IDFCorePlugin.getPlugin(), IDFCorePlugin.errorStatus("Status can't be null", null)); //$NON-NLS-1$
+
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java:114]: Can we `return` after logging the message

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java [hunk 3]
```diff
@@ -4,37 +4,125 @@
...
+			List<String> commands = new ArrayList<>();
+			commands.add(IDFUtil.getIDFPythonEnvPath());
+			commands.add(IDFUtil.getIDFPythonScriptFile().getAbsolutePath());
+			commands.add("--version"); //$NON-NLS-1$
+			Map<String, String> envMap = new IDFEnvironmentVariables().getEnvMap();
+			return runCommand(commands, envMap);
+		}
+		
+		return ""; //$NON-NLS-1$
+	}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java:102]: Can we move `getEspIdfVersion(..)` method to `IDFUtil`?

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
Thanks @sigmaaa for the PR. LGTM

**Comment by kolipakakondal** [comment]
Could you check build errors, seems to be build is getting failed due to API access restriction with the latest eclipse

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java]: changed. Thanks

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java (hunk 2 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java]: done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeConstants.java]: done

**Pull request merged**